openedx/core/djangoapps/dark_lang/tests.py

```diff
@@ -253,77 +256,32 @@ def _post_clear_preview_lang(self):
         """
         return self.client.post('/update_lang/', {'action': 'reset_preview_language'})
 
-    def _set_client_session_language(self, session_language):
-        """
-        Set the session language in the Client
-        """
-        session = self.client.session
-        session[LANGUAGE_SESSION_KEY] = session_language
-        session.save()
-
     def test_preview_lang_with_released_language(self):
+        self._set_client_cookie_language('rel')
         # Preview lang should always override selection
         self._post_set_preview_lang('rel')
         # Refresh the page with a get request to confirm the preview language was set
         self.client.get('/home')
-        self.assert_session_lang_equals(
-            'rel',
-            self.client.session
-        )
-
-        # Set the session language and ensure that the preview language overrides
-        self._set_client_session_language('notrel')
-        self._post_set_preview_lang('rel')
-        self.client.get('/home')
-        self.assert_session_lang_equals(
-            'rel',
-            self.client.session
-        )
+        self.assert_cookie_lang_equals('rel')
 
     def test_preview_lang_with_dark_language(self):
+        self._set_client_cookie_language('unrel')
```

Comment:
Same question. The need for this makes it seem like `self._post_set_preview_lang('unrel')` is no longer doing what it was supposed to be doing.